lib/ansible/modules/windows/win_dhcp_lease.ps1

```diff
@@ -0,0 +1,473 @@
+            $state_absent_removed = $false
+        }
+    }
+    else {
+
+        # Try to remove lease
+        Try {
+            $current_lease | Remove-DhcpServerv4Lease -WhatIf:$check_mode
+            $state_absent_removed = $true
+        }
+        Catch {
+            $state_absent_removed = $false
+        }
+    }
+
+    # If the lease doesn't exist, our work here is done
+    if ($current_lease_exists -eq $false) {
+        $module.Result.changed = $false
+        $module.Result.msg = "The lease or reservation doesn't exist."
+        $module.ExitJson()
+    }
+
+    # See if we removed the lease/reservation
+    if ($state_absent_removed) {
+        $module.Result.changed = $true
+        $module.ExitJson()
+    }
+    else {
+        $module.Result.lease = Convert-ReturnValue -Object $current_lease
+        $module.FailJson("Unable to remove lease/reservation")
+    }
+}
+
+# State: Present
+# Ensure the DHCP Lease/Reservation is present, and consistent
+if ($state -eq "present") {
+
+    # Current lease exists, and is not a reservation
+    if (($current_lease_reservation -eq $false) -and ($current_lease_exists -eq $true)) {
+        if ($type -eq "reservation") {
+            Try {
+                # Update parameters
+                $params = @{ }
+
+                if ($mac) {
+                    $params.ClientId = $mac
+                } else {
+                    $params.ClientId = $current_lease.ClientId
+                }
+
+                if ($description) {
+                    $params.Description = $description
+                } else {
+                    $params.Description = $current_lease.Description
+                }
+
+                if ($reservation_name) {
+                    $params.Name = $reservation_name
+                } else {
+                    $params.Name = "reservation-" + $params.ClientId
+                }
+
+                # Desired type is reservation
+                $current_lease | Add-DhcpServerv4Reservation -WhatIf:$check_mode
+
+                if(-not $check_mode) {
+                    $current_reservation = Get-DhcpServerv4Lease -ClientId $params.ClientId -ScopeId $current_lease.ScopeId
+                }
+
+                # Update the reservation with new values
+                $current_reservation | Set-DhcpServerv4Reservation @params -WhatIf:$check_mode
+
+                if(-not $check_mode) {
+                    $updated_reservation = Get-DhcpServerv4Lease -ClientId $params.ClientId -ScopeId $current_reservation.ScopeId
+                }
+
+                if(-not $check_mode) {
+                    # Compare Values
+                    $module.Result.changed = Compare-DhcpLease -Original $original_lease -Updated $updated_reservation
+                    $module.Result.lease = Convert-ReturnValue -Object $updated_reservation
+                } else {
+                    $module.Result.changed = $true
+                }
+
+                $module.ExitJson()
```

Comment:
Try to avoid calling ExitJson() in the middle of the code. Structure your code so that it can continue onto the end correctly.